### ISSUE TITLE
And then I will.

### ISSUE THREAD

**Issue by Contraculto** [dariusk/NaNoGenMo#70]
I have been experimenting with twitter searches for a couple of weeks after a little script I had to do for a client, and this gave me the idea of using that to recursively form a long text.

The idea is to have something not so random made from short phrases like "And then I will" or "And then I might". After dedicating most of my afternoon to the project I'd say I'm half way there.

**Comment by Contraculto** [comment]
Things are going great. The PHP script takes several minutes, but generates quite an interesting output. I need to filter and clean it a lot, but right now my searches create a "novel" of over 60.000 words so I'm quite confident it will be reaching all the goals.

Here is a sample of the dirty dirty output. https://gist.github.com/Contraculto/7536616

**Comment by dariusk** [comment]
Oh this is GREAT.

**Comment by enkiv2** [comment]
This is glorious. Are you pulling from the public timeline?

**Comment by MichaelPaulukonis** [comment]
And then we were dirty dirty dirty.

**Comment by Contraculto** [comment]
Yes, it's a series of searches made using the public API, which then get shuffled. I will be posting the code later today.

**Comment by Contraculto** [comment]
And we are ready. Everything works fine, and the generated novel is actually quite interesting (for a very long piece of random text). There's a lot of tweaking that could be done in the form of limits, cleaning and sorting to make it way more random or structured, adding variety to the phrases used, etc. but I'm liking what it does up to this point.

Check out the example 276.160 word novel here: http://contraculto.com/then-i-will/

The code is very simple and I tried to keep it as clean as possible. I also commented everything I thought would be useful. I'd love to see what others can do with that. Get it or fork it in it's repo: https://github.com/Contraculto/then-i-will

I hope at least someone enjoys playing with this as much as I enjoyed writing it.

**Comment by lilinx** [comment]
Very inspiring.

**Comment by ghost** [comment]
Very clever use of crowd sourced text to randomly generate a novel!   And if one considers the main character to be the 'I' in each statement ... sometimes it flows, sometimes it breaks down ... but what if the novel were the internal ramblings of a mad man or woman!  And the text is simply the out pouring of inner monologue ...

**Comment by ianrenton** [comment]
The first couple of lines of the example already made me laugh out loud!
"Then I will take no more physical, not even opiates, for I have prayed that I may render my soul to god unclouded.-horizon. Then maybe I will."

**Comment by Contraculto** [comment]
@ianrenton I generated lots of those when I was testing the script, and sometimes it just got very dark and depressing for a couple of lines, then crazy, happy again and so on. I had a great time with this.
